### PR TITLE
[reland] Enable assignment for QTensor in pytorch frontend

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#19676 [reland] Enable assignment for QTensor in pytorch frontend**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15064710/)

Make copy work with QTensor, enable assignment of QTensor in pytorch frontend.

Differential Revision: [D15064710](https://our.internmc.facebook.com/intern/diff/D15064710/)

### CODE REVIEW

File: aten/src/ATen/native/quantized/Copy.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#include <ATen/quantized/Quantizer.h>
+#include <ATen/CPUApplyUtils.h>
+
+namespace at { namespace native {
+Tensor& _s_copy__quantized(
+    Tensor& self,
+    const Tensor& src,
+    bool non_blocking) {
+  AT_CHECK(self.scalar_type() == at::kQInt8, "Quantized copy only works with kQInt8 as target Tensor");
+  AT_CHECK(src.scalar_type() == at::kFloat, "Quantized copy only works with kFloat as source Tensor");
```
Comment (zafartahirov) [inline aten/src/ATen/native/quantized/Copy.cpp]: Should the support be only for float, or int32 as well?

File: aten/src/ATen/native/quantized/Copy.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  qint8* self_data = self.data<qint8>();
+  float* src_data = src.data<float>();
+  for (int i = 0; i < self.numel(); ++i) {
```
Comment (zafartahirov) [inline aten/src/ATen/native/quantized/Copy.cpp]: I think we can use a tensor iterator here to speed things up

**Review by z-a-f — APPROVED** [review]
LGTM

File: aten/src/ATen/native/quantized/Copy.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    self_data[i] = quantize_uint8(self.q_scale().to<float>(), self.q_zero_point().to<uint8_t>(), src_data[i]);
```
Comment (ZolotukhinM) [inline aten/src/ATen/native/quantized/Copy.cpp]: Please clang-format before committing.

File: aten/src/ATen/native/quantized/Copy.cpp (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/native/quantized/Copy.cpp]: we will just support float32

File: aten/src/ATen/native/quantized/Copy.cpp (hunk 2 above)
Comment (jerryzh168) [reply aten/src/ATen/native/quantized/Copy.cpp]: I tired, but looks like TensorIterator calls type promotion which we didn't define for qint8

**Pull request closed without merging**